backend/src/load.rs

```diff
@@ -0,0 +1,404 @@
+        let repo_loc = Path::new("../data");
+
+        let mut last_date = None;
+        let mut crate_list = HashSet::new();
+        let mut phase_list = HashSet::new();
+        let mut no_times = 0;
+        let mut c_bootstraps = 0;
+        let mut c_benchmarks = 0;
+        let mut c_benchmarks_add = 0;
+
+        let mut data_rustc = Vec::new();
+        let mut data_benchmarks = Vec::new();
+        let mut benchmarks = HashSet::new();
+
+        // Read all files from repo_loc/processed
+        let mut file_count = 0;
+        for entry in try!(fs::read_dir(repo_loc.join("processed"))) {
+            let entry = try!(entry);
+            if try!(entry.metadata()).is_dir() {
+                continue;
+            }
+            file_count += 1;
+
+            let filename = entry.file_name().into_string().unwrap();
+            let mut file = try!(File::open(entry.path()));
+            let mut file_contents = String::new();
+            match try!(file.read_to_string(&mut file_contents)) {
+                // Skip files whose size is 0.
+                0 => {
+                    no_times += 1;
+                    continue;
+                },
+                _ => {}
+            }
+
+            let contents = try!(json::parse(&file_contents));
+            if contents["times"].is_empty() {
+                no_times += 1;
+                continue;
+            }
+
+            /// Parse date from JSON header in file contents.
+            fn parse_from_header(date: &str) -> Result<NaiveDateTime> {
+                // TODO: Determine correct format of header date and move into
+                // variable/constant
+                NaiveDateTime::parse_from_str(date, "%c %z")
+                    .map_err(|err| err.into())
+            }
+
+            /// Parse date from filename.
+            fn parse_from_filename(filename: &str) -> Result<NaiveDateTime> {
+                let date_str = &filename[(filename.find("--").unwrap() + 2)..filename.find(".json").unwrap()];
+
+                // Ignore seconds: they aren't key to the data processing, and
+                // parsing them requires logic to replace them with 0 when
+                // they are absent.
+                NaiveDateTime::parse_from_str(date_str, "%Y-%m-%d-%H-%M")
+                    .map_err(|err| err.into())
+            }
+
+            let header = &contents["header"];
+            let date = header["date"].as_str().unwrap();
+            let date = try!(parse_from_header(&date).or_else(|_| parse_from_filename(&filename)));
+
+            let test_name = &filename[0..filename.find("--").unwrap()];
+
+            let times = &contents["times"];
+            if test_name == "rustc" {
+                c_bootstraps += 1;
+                data_rustc.push(Data::new(date, &header, times, true));
+
+                for timing in times.members() {
+                    let crate_name = timing["crate"].as_str().unwrap().to_string();
```

Comment:
timing["crate"] is a [`JsonValue`](http://terhix.com/doc/json/enum.JsonValue.html); so to convert it to a string we need to take a `&str` out, unwrap that (it could be something else) and then convert it to a `String`.

This is one of the places where Serde might help... I'm still split on which one is better (Serde's ergonomics are terrible IMO if you don't use the Serialize/Deserialize auto-impls).